File: Lab6/Question7/src/Rectangle.java
public class Rectangle extends Shape{
    private double width;
    private double length;

    public Rectangle(double x, double y, double width, double length) {
        super(x, y);
        this.width = width;
        this.length = length;
    }

    @Override
    public double area() {
        return width * length;
    }

    public static void testRectangle() {
        Rectangle rectangle = new Rectangle(2, 1, 3, 4);
        System.out.println(rectangle.getX() == 2);
        System.out.println(rectangle.getY() == 1);
        System.out.println(rectangle.area() == 12);
    }
    
    @Override
    public String toString() {
        return this.getClass().getName() + " has area " + area();
    }
}

File: Lab6/Rectangle.kt
open class Rectangle : Shape {
    private val width: Double
    private val length: Double

    constructor(x: Double, y: Double, width: Double, length: Double) : super(x, y) {
        this.width = width
        this.length = length
    }

    override fun area(): Double {
        return width * length
    }

    override fun toString(): String {
        return "${this.javaClass.name} has area ${area()}"
    }

    companion object {
        fun testRectangle() {
            val rectangle = Rectangle(2.0, 1.0, 3.0, 4.0)
            println(rectangle.getX() == 2.0)
            println(rectangle.getY() == 1.0)
            println(rectangle.area() == 12.0)
        }
    }
}